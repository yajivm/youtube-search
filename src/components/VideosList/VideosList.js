import React from 'react';
import PropTypes from 'prop-types';
import VideoTile from '../VideoTile/VideoTile';

import './VideosList.style.css';

const VideosList = ({ listData, onVideoSelected, isThumbnailLayout }) => {

    const getVideosList = () => {
      if (listData?.length) {
        return listData.map((videoData, index) => {
          return (
            <div key={index} data-testid="video-thumbnail-wrapper" className={isThumbnailLayout ? 'thumbnail-wrapper' : 'video-wrapper'}>
              <VideoTile videoData={videoData} onSelectVideoTile={onVideoSelected} isThumbnailLayout={isThumbnailLayout} />
            </div>
          );
        });
      } else {
        return <></>;
      }
    };

    return (
        <>{getVideosList()}</>
    );
};

VideosList.propTypes = {
  listData: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  onVideoSelected: PropTypes.func.isRequired,
  isThumbnailLayout: PropTypes.bool,
};

VideosList.defaultProps = {
  isThumbnailLayout: false,
};

export default VideosList;
