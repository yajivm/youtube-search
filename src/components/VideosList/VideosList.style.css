.video-wrapper {
    width: 25%;
}