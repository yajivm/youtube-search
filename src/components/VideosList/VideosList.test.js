import React from 'react';
import { render } from '@testing-library/react';

import VideosList from './VideosList';

jest.mock('../VideoTile/VideoTile', () => (props) => <div data-testid="video-thumbnail" {...props} />);

let mockListData = [{ test: 'testData' }];
const mockOnVideoSelected = jest.fn();

test('should render video list component', () => {
    const { getByTestId } = render(<VideosList listData={mockListData} onVideoSelected={mockOnVideoSelected} />);
    
    const videoThumbnailWrapper = getByTestId('video-thumbnail-wrapper');

    expect(videoThumbnailWrapper).toBeDefined();
    expect(videoThumbnailWrapper).toMatchSnapshot();
});