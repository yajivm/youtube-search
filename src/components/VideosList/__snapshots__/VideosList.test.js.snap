// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render video list component 1`] = `
<div
  class="video-wrapper"
  data-testid="video-thumbnail-wrapper"
>
  <div
    data-testid="video-thumbnail"
    videodata="[object Object]"
  />
</div>
`;
